File: CF Python/K.py
def sum_differences(nums: list):
    l = len(nums)
    coef = range(-(l - 1), l, 2)
    s = 0
    for i in range(l):
        s += coef[i] * nums[i]

    return s * 2


k = int(input())
n = int(input())
nums_in_cat = {y: [] for y in range(1, k + 1)}
all_nums = []

for i in range(n):
    x, y = map(int, input().split())
    nums_in_cat[y].append(x)
    all_nums.append(x)

all_nums.sort()
for i in range(1, k + 1):
    nums_in_cat[i].sort()

whole = sum_differences(all_nums)
inner = 0
for nums in nums_in_cat.values():
    inner += sum_differences(nums)

print(inner)
print(whole - inner)

File: CF PABC.NET/K.pas
function SumDifferences(Self: List<Int64>): Int64; extensionmethod;
begin
  var L := Self.Count;
  Result := Range(-L + 1, L + 1, 2).Zip(Self, (Coef, I) -> Coef * I).Sum() * 2;
end;

begin
  var (K, N) := ReadlnInteger2();
  var NumsInCat := Range(1, K).ToDictionary(K -> K, V -> new List<Int64>);
  var AllNums := new List<Int64>;
  
  loop N do
  begin
    var (X, Y) := ReadlnInteger2();
    NumsInCat[Y].Add(X);
    AllNums.Add(X);
  end;
  
  AllNums.Sort();
  for var I := 1 to K do
    NumsInCat[I].Sort();
  
  var Whole := AllNums.SumDifferences();
  var Inner := NumsInCat.Values.Select(Cat -> Cat.SumDifferences()).Sum();
  Inner.Println();
  (Whole - Inner).Println();
end.